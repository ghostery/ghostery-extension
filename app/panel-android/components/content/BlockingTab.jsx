/**
 * Blocking Tab Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import DotsMenu from './DotsMenu';
import BlockingCategories from './BlockingCategories';

class BlockingTab extends React.Component {
	constructor(props) {
		super(props);

		const { callGlobalAction } = props;
		this.siteActions = [
			{
				id: 'blockAllSite',
				name: t('blocking_block_all'),
				callback: () => {
					callGlobalAction({
						actionName: 'blockUnBlockAllTrackers',
						actionData: { block: true, type: 'site' }
					});
				}
			}, {
				id: 'unblockAllSite',
				name: t('blocking_unblock_all'),
				callback: () => {
					callGlobalAction({
						actionName: 'blockUnBlockAllTrackers',
						actionData: { block: false, type: 'site' }
					});
				}
			}
		];
		this.globalActions = [
			{
				id: 'blockAllGlobal',
				name: t('blocking_block_all'),
				callback: () => {
					callGlobalAction({
						actionName: 'blockUnBlockAllTrackers',
						actionData: { block: true, type: 'global' }
					});
				}
			}, {
				id: 'unblockAllGlobal',
				name: t('blocking_unblock_all'),
				callback: () => {
					callGlobalAction({
						actionName: 'blockUnBlockAllTrackers',
						actionData: { block: false, type: 'global' }
					});
				}
			}, {
				id: 'resetSettings',
				name: t('android_blocking_reset'),
				callback: () => {
					callGlobalAction({
						actionName: 'resetSettings',
					});
				}
			}
		];
	}

	get actions() {
		const { type } = this.props;
		if (type === 'site') {
			return this.siteActions;
		}
		return this.globalActions;
	}

	get headerText() {
		const { type } = this.props;
		return (type === 'site') ?
			t('android_site_blocking_header') :
			t('android_global_blocking_header');
	}

	render() {
		const {
			type,
			categories,
			settings,
			siteProps,
			callGlobalAction,
		} = this.props;

		return (
			<div className="BlockingHeader">
				<div className="BlockingHeader__text">
					<h2>{this.headerText}</h2>
					<DotsMenu actions={this.actions} />
				</div>
				<BlockingCategories
					type={type}
					categories={categories}
					settings={settings}
					siteProps={siteProps}
					callGlobalAction={callGlobalAction}
				/>
			</div>
		);
	}
}

BlockingTab.propTypes = {
	type: PropTypes.oneOf([
		'site',
		'global',
	]).isRequired,
	callGlobalAction: PropTypes.func.isRequired,
	siteProps: PropTypes.shape({}).isRequired,
	categories: PropTypes.arrayOf(PropTypes.shape({})),
	settings: PropTypes.shape({}),
};

BlockingTab.defaultProps = {
	categories: [],
	settings: {},
};

export default BlockingTab;
