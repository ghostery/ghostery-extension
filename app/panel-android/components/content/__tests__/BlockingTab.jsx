/**
 * BlockingTab Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import BlockingTab from '../BlockingTab';

describe('app/panel-android/components/content/BlockingTab.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('BlockingTab component as site with falsy props', () => {
			const component = renderer.create(
				<BlockingTab
					type="site"
					categories={[]}
					siteProps={{}}
					callGlobalAction={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('BlockingTab component as global with falsy props', () => {
			const component = renderer.create(
				<BlockingTab
					type="global"
					categories={[]}
					siteProps={{}}
					callGlobalAction={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('BlockingTab component as site with tracker falsy props', () => {
			const component = renderer.create(
				<BlockingTab
					type="site"
					categories={[
						{
							id: '1',
							name: 'Test1',
							num_total: 1,
							num_blocked: 1,
							trackers: [],
							img_name: 'category-image-url',
						},
					]}
					siteProps={{
						isTrusted: false,
						isRestricted: false,
						isPaused: false,
					}}
					callGlobalAction={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('BlockingTab component as global with tracker truthy props', () => {
			const component = renderer.create(
				<BlockingTab
					type="global"
					categories={[
						{
							id: '1',
							name: 'Test1',
							num_total: 1,
							num_blocked: 1,
							trackers: [],
							img_name: 'category-image-url',
						},
						{
							id: '2',
							name: 'Test2',
							num_total: 5,
							num_blocked: 3,
							trackers: [],
							img_name: 'category-image-url-2',
						},
					]}
					siteProps={{
						isTrusted: true,
						isRestricted: false,
						isPaused: true,
					}}
					callGlobalAction={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});
});
