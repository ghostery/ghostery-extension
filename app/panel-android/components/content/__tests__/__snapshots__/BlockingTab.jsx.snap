// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel-android/components/content/BlockingTab.jsx Snapshot tests with react-test-renderer BlockingTab component as global with falsy props 1`] = `
<div
  className="BlockingHeader"
>
  <div
    className="BlockingHeader__text"
  >
    <h2>
      android_global_blocking_header
    </h2>
    <div
      className="DotsMenu"
    >
      <button
        aria-label="Menu"
        className="DotsMenu__button"
        onClick={[Function]}
        type="button"
      />
      <div
        className="DotsMenu__content"
      >
        <ul>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_block_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_unblock_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              android_blocking_reset
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="BlockingCategories"
  />
</div>
`;

exports[`app/panel-android/components/content/BlockingTab.jsx Snapshot tests with react-test-renderer BlockingTab component as global with tracker truthy props 1`] = `
<div
  className="BlockingHeader"
>
  <div
    className="BlockingHeader__text"
  >
    <h2>
      android_global_blocking_header
    </h2>
    <div
      className="DotsMenu"
    >
      <button
        aria-label="Menu"
        className="DotsMenu__button"
        onClick={[Function]}
        type="button"
      />
      <div
        className="DotsMenu__content"
      >
        <ul>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_block_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_unblock_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              android_blocking_reset
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="BlockingCategories"
  >
    <div
      className="BlockingCategory"
    >
      <div
        className="BlockingCategory__details flex-container"
        onClick={[Function]}
      >
        <img
          className="BlockingCategory__image"
          src="/app/images/panel-android/categories/category-image-url.svg"
        />
        <div
          className="flex-child-grow"
        >
          <h2
            className="BlockingCategory__name"
          >
            Test1
          </h2>
          <div>
            <span
              className="BlockingCategory__numTrackers"
            >
              1 blocking_category_tracker
            </span>
            <span
              className="BlockingCategory__numBlocked"
            >
              1 blocking_category_blocked
            </span>
          </div>
        </div>
        <div
          className="BlockingCategory__buttons flex-container flex-dir-column align-justify align-middle"
        >
          <div
            className="BlockingCategory--noPointer"
          >
            <div
              className="BlockingSelectButton BlockingSelectButton__blocked"
              onClick={[Function]}
            />
          </div>
          <div
            className="BlockingCategory__toggle"
          />
        </div>
      </div>
      <div
        className="BlockingCategory__list"
        style={
          Object {
            "height": 0,
          }
        }
      />
    </div>
    <div
      className="BlockingCategory"
    >
      <div
        className="BlockingCategory__details flex-container"
        onClick={[Function]}
      >
        <img
          className="BlockingCategory__image"
          src="/app/images/panel-android/categories/category-image-url-2.svg"
        />
        <div
          className="flex-child-grow"
        >
          <h2
            className="BlockingCategory__name"
          >
            Test2
          </h2>
          <div>
            <span
              className="BlockingCategory__numTrackers"
            >
              5 blocking_category_trackers
            </span>
            <span
              className="BlockingCategory__numBlocked"
            >
              3 blocking_category_blocked
            </span>
          </div>
        </div>
        <div
          className="BlockingCategory__buttons flex-container flex-dir-column align-justify align-middle"
        >
          <div
            className="BlockingCategory--noPointer"
          >
            <div
              className="BlockingSelectButton BlockingSelectButton__mixed"
              onClick={[Function]}
            />
          </div>
          <div
            className="BlockingCategory__toggle"
          />
        </div>
      </div>
      <div
        className="BlockingCategory__list"
        style={
          Object {
            "height": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`app/panel-android/components/content/BlockingTab.jsx Snapshot tests with react-test-renderer BlockingTab component as site with falsy props 1`] = `
<div
  className="BlockingHeader"
>
  <div
    className="BlockingHeader__text"
  >
    <h2>
      android_site_blocking_header
    </h2>
    <div
      className="DotsMenu"
    >
      <button
        aria-label="Menu"
        className="DotsMenu__button"
        onClick={[Function]}
        type="button"
      />
      <div
        className="DotsMenu__content"
      >
        <ul>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_block_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_unblock_all
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="BlockingCategories"
  />
</div>
`;

exports[`app/panel-android/components/content/BlockingTab.jsx Snapshot tests with react-test-renderer BlockingTab component as site with tracker falsy props 1`] = `
<div
  className="BlockingHeader"
>
  <div
    className="BlockingHeader__text"
  >
    <h2>
      android_site_blocking_header
    </h2>
    <div
      className="DotsMenu"
    >
      <button
        aria-label="Menu"
        className="DotsMenu__button"
        onClick={[Function]}
        type="button"
      />
      <div
        className="DotsMenu__content"
      >
        <ul>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_block_all
            </button>
          </li>
          <li>
            <button
              className="DotsMenu__item"
              onClick={[Function]}
              type="button"
            >
              blocking_unblock_all
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="BlockingCategories"
  >
    <div
      className="BlockingCategory"
    >
      <div
        className="BlockingCategory__details flex-container"
        onClick={[Function]}
      >
        <img
          className="BlockingCategory__image"
          src="/app/images/panel-android/categories/category-image-url.svg"
        />
        <div
          className="flex-child-grow"
        >
          <h2
            className="BlockingCategory__name"
          >
            Test1
          </h2>
          <div>
            <span
              className="BlockingCategory__numTrackers"
            >
              1 blocking_category_tracker
            </span>
            <span
              className="BlockingCategory__numBlocked"
            >
              1 blocking_category_blocked
            </span>
          </div>
        </div>
        <div
          className="BlockingCategory__buttons flex-container flex-dir-column align-justify align-middle"
        >
          <div
            className="BlockingCategory--noPointer"
          >
            <div
              className="BlockingSelectButton BlockingSelectButton__trusted"
              onClick={[Function]}
            />
          </div>
          <div
            className="BlockingCategory__toggle"
          />
        </div>
      </div>
      <div
        className="BlockingCategory__list"
        style={
          Object {
            "height": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
